sktime/classification/dictionary_based/_tde.py

```diff
@@ -0,0 +1,417 @@
+        self.igb_options = [True, False]
+        super(TemporalDictionaryEnsemble, self).__init__()
+
+    def fit(self, X, y):
+        """Build an ensemble of individual TDE classifiers from the training
+        set (X,y), through randomising over the parameter space to a set
+        number of times then selecting new parameters using Gaussian
+        processes
+
+        Parameters
+        ----------
+        X : nested pandas DataFrame of shape [n_instances, 1]
+            Nested dataframe with univariate time-series in cells.
+        y : array-like, shape = [n_instances] The class labels.
+
+        Returns
+        -------
+        self : object
+        """
+
+        X, y = check_X_y(X, y, enforce_univariate=True)
+        y = y.values if isinstance(y, pd.Series) else y
+
+        self.time_limit = self.time_limit * 60
+        self.n_instances, self.series_length = X.shape[0], len(X.iloc[0, 0])
+        self.n_classes = np.unique(y).shape[0]
+        self.classes_ = class_distribution(np.asarray(y).reshape(-1, 1))[0][0]
+        for index, classVal in enumerate(self.classes_):
+            self.class_dictionary[classVal] = index
+
+        self.classifiers = []
+        self.weights = []
+        self.prev_parameters_x = []
+        self.prev_parameters_y = []
+
+        # Window length parameter space dependent on series length
+        max_window_searches = self.series_length / 4
+        max_window = int(self.series_length * self.max_win_len_prop)
+        win_inc = int((max_window - self.min_window) / max_window_searches)
+        if win_inc < 1:
+            win_inc = 1
+
+        possible_parameters = self._unique_parameters(max_window, win_inc)
+        num_classifiers = 0
+        start_time = time.time()
+        train_time = 0
+        subsample_size = int(self.n_instances * 0.7)
+        lowest_acc = 0
+        lowest_acc_idx = 0
+
+        if self.time_limit > 0:
+            self.n_parameter_samples = 0
+
+        rng = check_random_state(self.random_state)
+
+        while (train_time < self.time_limit or num_classifiers <
+               self.n_parameter_samples) and len(possible_parameters) > 0:
+            if num_classifiers < self.randomly_selected_params:
+                parameters = possible_parameters.pop(
+                    rng.randint(0, len(possible_parameters)))
+            else:
+                gp = GaussianProcessRegressor(random_state=self.random_state)
+                gp.fit(self.prev_parameters_x, self.prev_parameters_y)
+                preds = gp.predict(possible_parameters)
+                parameters = possible_parameters.pop(rng.choice(
+                    np.flatnonzero(preds == preds.max())))
+
+            subsample = rng.choice(self.n_instances, size=subsample_size,
+                                   replace=False)
+            X_subsample = X.iloc[subsample, :]
+            y_subsample = y[subsample]
+
+            tde = IndividualTDE(parameters[0], parameters[1],
```

Comment:
Could you not use `**parameters` here to unnest and pass dictionary as keyword arguments to the constructor?

Reply:
Made a change making use of `*parameters` to clean it a bit. The parameter lists are fed to the GP  so cant be a dictionary (though now that i say that, i could have just extracted the values when they are added to the previous parameter list).